### DISCUSSION TITLE
Database

### DISCUSSION THREAD

**Discussion by SkyDude0** [games647/FastLogin#787] [General]
Can i use MYSQL on velocity proxy? when i try it dosent work with error ''java.sql.SQLException: Client does not support authentication protocol requested by server. plugin type was = 'mysql_native_password' ''

If as i imagine is only with mariadb, is possible to have mysql in the future? is the only pl that dosent have the choice

**Comment by TuxCoding** [comment]
The MariaDB connector supports both authentication methods ([1]), including `mysql_native_password`. Do you have another MySQL/MariaDB connector installed? Could you maybe post the full stacktrace.

[1] https://mariadb.com/kb/en/about-mariadb-connector-j/#server-compatibility

**Reply by SkyDude0** [reply]
you mean this? https://pastebin.com/FMqmgQRS

i have mysql 8, how can i see witch connector i use?

Also.. if i use waterfall with the other driver it work

**Reply by TuxCoding** [reply]
Looks like MySQL 8 uses a newer authentication method. I don't think `mysql_native_password` is used here, but it's rather a translation error. Could you look up which authentication method is configured for your database user. Maybe set the old method using `ALTER USER ... IDENTIFIED ....` and `flush ...`.

EDIT: Or force to the newer one `caching_sha2_password`

**Reply by SkyDude0** [reply]
i use this command for mysql_native_password 

ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY 'password';

..

Tomorrow ill try do edit the file my.cnf to change default config.. the default are already with: caching_sha2_password

**Reply by SkyDude0** [reply]
solved
